pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/model/config/CastConfig.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.model.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
+import lombok.Getter;
+
+@Getter
+public class CastConfig extends StepConfig {
+  @JsonProperty(value = "schema-type", required = true)
+  private String schemaType;
+
+  private Optional<String> part = Optional.empty();
```

Comment:
yeah I remember this from another feedback you gave, I do agree that optionals are anti-pattern for business logics classes, but for config models only stuff I found them useful because they communicate clearly that this part of the config is optional with default value equal to null so the person using this config will not forget to handle nulls.

However they don't add huge value in our case (and always good to remove dependency!) - I'll remove it.